src/webgpu/api/validation/capability_checks/limits/maxFragmentCombinedOutputResources.spec.ts

```diff
@@ -0,0 +1,145 @@
+import { range } from '../../../../../common/util/util.js';
+
+import {
+  computeBytesPerSample,
+  kMaximumLimitBaseParams,
+  makeLimitTestGroup,
+  LimitsRequest,
+} from './limit_utils.js';
+
+function select<T>(array: T[], i: number) {
+  return array[i % array.length];
+}
+
+const kTextureStorageTypes = [
+  { type: 'texture_storage_1d', coords: '0' },
+  { type: 'texture_storage_2d', coords: 'vec2u(0)' },
+  { type: 'texture_storage_2d_array', coords: 'vec2u(0), 0' },
+  { type: 'texture_storage_3d', coords: 'vec3u(0)' },
+];
+
+function getPipelineDescriptor(device: GPUDevice, testValue: number) {
+  const numStorageTextures = (testValue / 2) | 0;
```

Comment:
So for the limits here, would be nice to fill it with both storage buffers and textures, leaving the rest for the color attachments. I wrote similar tests in `RenderPipelineValidationTests.cpp` here https://dawn-review.googlesource.com/c/dawn/+/122461 for reference. I think the `kExtraLimits` also probably wants to reference `maxStorageBuffersPerShaderStage` and `maxStorageTexturesPerShaderStage` as those limits are also relevant when trying to fill out the resource limit.

Reply:
done